### ISSUE TITLE
Cannot innit on Nixos

### ISSUE THREAD

**Issue by simon-bueler** [H3rmt/hyprswitch#71]
```bash
 ~#@❯ hyprswitch -vv init &
[1] 350303
INFO Starting Hyprswitch (3.0.1) on Hyprland                                                                                                                                                
Error: Unable to parse Hyprland Version

Caused by:
    empty string, expected a semver version
[1]  + exit 1     hyprswitch -vv init
 ~#@❯ Hyprland --version
Hyprland 0.44.0 built from branch  at commit 88e9e0394541a853600bc2c910005c05fa156269  ().
Date: 2024-11-04
Tag: , commits: 5410
built against aquamarine 0.4.3
```

Not sure what the problem is here.

**Comment by H3rmt** [comment]
Ok, that's confusing.

Hyprswitch checks the running hyprland version to prevent weird bugs from happening when running outdated hyprland.

Can you you run `hyprctl version -j`

**Comment by simon-bueler** [comment]
```bash
 ~#@❯ hyprctl version -j
{
    "branch": "",
    "commit": "0fb9a04526b06adfb3fd16b64e13b7d110ae7855",
    "version": "0.44.0",
    "dirty": false,
    "commit_message": "",
    "commit_date": "2024-11-04",
    "tag": "",
    "commits": "5409",
    "buildAquamarine": "0.4.3",
    "flags": ["debug"]
}
```

**Comment by H3rmt** [comment]
I think I know why, the hyprland version is parsed from the Tag which is missing for the hyprland-git variant.

**Comment by H3rmt** [comment]
Using the version instead of the Tag would definitely make more sense

**Comment by H3rmt** [comment]
I will push a fix release asap

**Comment by simon-bueler** [comment]
Seems like the cargo hyprland package `pub struct Version` is missing the version field.

**Comment by H3rmt** [comment]
I updated the version struct like 2 weeks ago, pretty sure the version field wasn't there...

**Comment by yavko** [comment]
> I updated the version struct like 2 weeks ago, pretty sure the version field wasn't there...

Bruh, well feel free to PR it in

**Comment by simon-bueler** [comment]
fixed with v3.0.2
